project_1/src/main/java/app/controller/client/OrderController.java

```diff
@@ -0,0 +1,44 @@
+package app.controller.client;
+
+import app.controller.BaseController;
+import app.model.Cart;
+import app.model.CartDetail;
+import app.model.Order;
+import app.model.OrderDetail;
+import org.springframework.mail.SimpleMailMessage;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Controller
+@RequestMapping(value = "/orders")
+public class OrderController extends BaseController {
+
+    @RequestMapping(value = "/purchase", method = RequestMethod.POST)
+    public String purchase(@RequestParam Integer id) {
+        Cart cart = cartService.findById(id);
+        Order order = new Order(cart);
+        List<OrderDetail> orderDetails = new ArrayList<>();
+        for (CartDetail cartDetail : cart.getCartDetails()) {
+            if (cartDetail != null) {
+                orderDetails.add(new OrderDetail(cartDetail));
+            }
+        }
+        if (orderService.saveOrderAndRemoveCart(order, orderDetails, cart)) {
+            sendSimpleMessage(cart.getUser().getEmail(), "Hello " + cart.getUser().getFullName() + "!", "Thank you for believing us! ");
```

Comment:
`"Hello " + cart.getUser().getFullName() + "!", "Thank you for believing us! "`
Khai báo vào file messages.properties nào, trong file messages có thể dùng params dc nhé, chứ ko phải nối chuỗi mô